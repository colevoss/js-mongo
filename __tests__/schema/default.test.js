const Schema = require('../../src/Schema');

describe('Schema.<type>.default', () => {
  test('Returns the provided default value for the given key', () => {
    const testData = { notTest: 1 };

    const validated = Schema.number.default(1)('test', testData);

    expect(validated).toBe(1);
  });

  test('Returns value for provided defaulter function for the given key', () => {
    const testData = { notTest: 1 };

    const validated = Schema.number.default(() => 1)('test', testData);

    expect(validated).toBe(1);
  });

  test('Returns error if default value does not match schema type', () => {
    const testData = { notTest: 1 };

    const validated = Schema.number.default('string')('test', testData);

    expect(validated).toBeInstanceOf(TypeError);
    expect(validated).toMatchSnapshot();
  });
});
