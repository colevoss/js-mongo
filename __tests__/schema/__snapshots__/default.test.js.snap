// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.<type>.default Returns error if default value does not match schema type 1`] = `[TypeError: Expected test to be of type number but recieved string of string]`;
